framework/src/modules/chain/blocks/blockV2.js

```diff
@@ -0,0 +1,471 @@
+			delete block[key];
+		}
+	});
+	try {
+		validator.validate(blockSchema, block);
+	} catch (schemaError) {
+		throw schemaError.errors;
+	}
+	const { transactionsResponses } = validateTransactions(exceptions)(
+		block.transactions
+	);
+	const invalidTransactionResponse = transactionsResponses.find(
+		transactionResponse => transactionResponse.status !== TransactionStatus.OK
+	);
+	if (invalidTransactionResponse) {
+		throw invalidTransactionResponse.errors;
+	}
+	return block;
+};
+
+/**
+ * Verifies block hash, generator block publicKey and block signature.
+ *
+ * @param {block} block
+ * @throws {Error}
+ * @returns {boolean} Verified hash, signature and publicKey
+ * @todo Add description for the params
+ */
+const verifySignature = block => {
+	const signatureLength = 64;
+	const data = getBytesV2(block);
+	const dataWithoutSignature = Buffer.alloc(data.length - signatureLength);
+
+	for (let i = 0; i < dataWithoutSignature.length; i++) {
+		dataWithoutSignature[i] = data[i];
+	}
+	const hashedBlock = hash(dataWithoutSignature);
+	return verifyData(
+		hashedBlock,
+		block.blockSignature,
+		block.generatorPublicKey
+	);
+};
+
+/**
+ * Calculates block id based on block.
+ *
+ * @param {block} block
+ * @returns {string} Block id
+ * @todo Add description for the params
+ */
+const getId = block => {
+	const hashedBlock = hash(getBytesV2(block));
+	const temp = Buffer.alloc(8);
+	for (let i = 0; i < 8; i++) {
+		temp[i] = hashedBlock[7 - i];
+	}
+
+	// eslint-disable-next-line new-cap
+	const id = new BigNum.fromBuffer(temp).toString();
+	return id;
+};
+
+const blockSchema = {
+	type: 'object',
+	properties: {
+		id: {
+			type: 'string',
+			format: 'id',
+			minLength: 1,
+			maxLength: 20,
+		},
+		height: {
+			type: 'integer',
+		},
+		maxHeightPreviouslyForged: {
+			type: 'integer',
+		},
+		prevotedConfirmedUptoHeight: {
+			type: 'integer',
+		},
+		blockSignature: {
+			type: 'string',
+			format: 'signature',
+		},
+		generatorPublicKey: {
+			type: 'string',
+			format: 'publicKey',
+		},
+		numberOfTransactions: {
+			type: 'integer',
+		},
+		payloadHash: {
+			type: 'string',
+			format: 'hex',
+		},
+		payloadLength: {
+			type: 'integer',
+		},
+		previousBlock: {
+			type: 'string',
+			format: 'id',
+			minLength: 1,
+			maxLength: 20,
+		},
+		timestamp: {
+			type: 'integer',
+		},
+		totalAmount: {
+			type: 'object',
+			format: 'amount',
+		},
+		totalFee: {
+			type: 'object',
+			format: 'amount',
+		},
+		reward: {
+			type: 'object',
+			format: 'amount',
+		},
+		transactions: {
+			type: 'array',
+			uniqueItems: true,
+		},
+		version: {
+			type: 'integer',
+			minimum: 0,
+		},
+	},
+	required: [
+		'blockSignature',
+		'generatorPublicKey',
+		'numberOfTransactions',
+		'payloadHash',
+		'payloadLength',
+		'timestamp',
+		'totalAmount',
+		'totalFee',
+		'reward',
+		'transactions',
+		'version',
+	],
+};
+
+module.exports = {
+	getBytesV2,
```

Comment:
Since the file itself is `blocks_v2.js`, we should export the function names from there with out postfix, so simple `getBytes`, `create`, `dbRead`